Added documentation for Introspection API (#17753)
hashicorp/vault: website/content/api-docs/system/inspect/router.mdx

---
layout: api
page_title: /sys/internal/inspect/router - HTTP API
description: >-
  The '/sys/internal/inspect/router' endpoint focuses on viewing the contents of specific structures in the internal router subsystem.
---

# `/sys/internal/inspect/router`

> The `/sys/internal/inspect/router` is intended for Vault admin to inspect a specific internal subsystem for debugging purposes.

Reviewer: ```suggestion
The `/sys/internal/inspect/router` is intended for a Vault admin to inspect a specific internal subsystem of the router for debugging purposes.
```

> The `/sys/internal/inspect/router` is intended for a Vault admin to inspect a specific internal subsystem for debugging purposes.

Reviewer: ```suggestion
The `/sys/internal/inspect/router` endpoint is intended for a Vault admin to inspect the internal components of Vault's router.
```

> This token can be accessed with a root token or sudo privileges.

Reviewer: ```suggestion
This endpoint can be accessed with a root token or sudo privileges.
```

Author: Done!

        "mount_namespace": "",
        "mount_path": "token/",
        "mount_type": "token",
        "storage_prefix": "sys/token/",
        "tainted": false,
        "uuid": "de61dd8c-a181-ccc2-457d-908e417d10b0"
      },
      {
        "accessor": "cubbyhole_58b6727b",
        "mount_namespace": "",
        "mount_path": "cubbyhole/",
        "mount_type": "cubbyhole",
        "storage_prefix": "logical/496bc2e4-b641-2561-1829-f6557bf1fedc/",
        "tainted": false,
        "uuid": "496bc2e4-b641-2561-1829-f6557bf1fedc"
      },
      {
        "accessor": "identity_2ccfb6ab",
        "mount_namespace": "",
        "mount_path": "identity/",
        "mount_type": "identity",
        "storage_prefix": "logical/b15e93e0-5bf1-07b0-86d3-563f9c77eef1/",
        "tainted": false,
        "uuid": "b15e93e0-5bf1-07b0-86d3-563f9c77eef1"
      },
      {
        "accessor": "system_0157e00a",
        "mount_namespace": "",
        "mount_path": "sys/",
        "mount_type": "system",
        "storage_prefix": "sys/",
        "tainted": false,
        "uuid": "8ac2615e-8a73-fe66-52a6-a422f7e3273f"
      }
    ]
  },
  "warnings": null
}
```

## Mount UUID Cache

> This endpoint returns a list of mount entries in the router's mount uuid cache.

Reviewer: ```suggestion
This endpoint returns a list of mount entries in the router's mount UUID cache.
```

---
layout: api
page_title: /sys/internal/inspect/router - HTTP API
description: >-
  The '/sys/internal/inspect/router' endpoint focuses on viewing the contents of specific structures in the internal router subsystem.
---

> # `/sys/raw`

Reviewer 2: ```suggestion
# `/sys/internal/inspect/router`
```

Reviewer 2: Might be worth grepping for "raw" on these new pages, though I didn't see it anywhere else.

> The `/sys/internal/inspect/router` is intended to inspect a specific internal subsytem for debugging purposes.
> It can be accessed with a root token or sudo priveleges.

Reviewer 2: ```suggestion
The `/sys/internal/inspect/router` is intended to inspect a specific internal subsystem for debugging purposes.
It can be accessed with a root token or sudo privileges.
```

## Root

This endpoint returns a list of router entries in the router's root tree.

| Method | Path          |
| :----- | :------------ |
| `GET`  | `/sys/internal/inspect/router/root` |

> ### Sample Request

Reviewer 2: I think it would be nice to include sample responses for these commands. The sample responses are really helpful when I'm trying to understand what a command actually shows.

Author: Added this information in the current commit!

Reviewer 2: Thanks!
